Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Sandboxed/SelectorsPollerScript.js

```diff
@@ -489,33 +491,33 @@ window.__firefox__.execute(function($) {
     }
     // Find selectors we have a rule index for
     const rulesToRemove = Array.from(selectors)
-      .map(function (selector) { return CC.allSelectorsToRules.get(selector) })
+      .map(function (selector) { return CC.allHideSelectorsToRules.get(selector) })
       .filter(function (i) { return i !== undefined })
       .sort()
       .reverse()
 
     // Delete the rules
-    const lastIdx = CC.allSelectorsToRules.size - 1
+    const lastIdx = CC.allHideSelectorsToRules.size - 1
     // to deal with removing rules
     
     for (let _i = 0, rulesToRemove1 = rulesToRemove; _i < rulesToRemove1.length; _i++) {
       const ruleIdx = rulesToRemove1[_i]
       // Safe to asset ruleIdx is a number because we've already filtered out
       // any `undefined` instances with the filter call above.
-      CC.cosmeticStyleSheet.deleteRule(ruleIdx)
+      delete myArray[ruleIdx]
```

Comment:
`delete` in JS causes all sorts of unexpected perf issues.  Probably better to rework this.  Also, probably better to use a more descripitve name than `myArray` ;)

Reply:
I need these indexes to stay in place for now because they are stored in a hash and are referenced for the non-aggressive mode. I will look at cleaning up this file but for now a lot of this logic is there to work on desktop and unless I start diverging this file from the desktop version and cleaning up what is not necessary for iOS it will be tricky to remove this delete.

Reply:
though myArray is a definite typo.

Reply:
Fixed